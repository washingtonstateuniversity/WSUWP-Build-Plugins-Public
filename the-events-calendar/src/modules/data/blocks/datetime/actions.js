/**
 * Internal dependencies
 */
import * as types from './types';

export const setNaturalLanguageLabel = ( label ) => ( {
	type: types.SET_NATURAL_LANGUAGE_LABEL,
	payload: {
		label,
	},
} );

export const setDateRange = ( payload ) => ( {
	type: types.SET_DATE_RANGE,
	payload,
} );

export const setStartDateTime = ( start ) => ( {
	type: types.SET_START_DATE_TIME,
	payload: {
		start,
	},
} );

export const setEndDateTime = ( end ) => ( {
	type: types.SET_END_DATE_TIME,
	payload: {
		end,
	},
} );

export const setStartTime = ( start ) => ( {
	type: types.SET_START_TIME,
	payload: {
		start,
	},
} );

export const setEndTime = ( end ) => ( {
	type: types.SET_END_TIME,
	payload: {
		end,
	},
} );

export const setStartTimeInput = ( startTimeInput ) => ( {
	type: types.SET_START_TIME_INPUT,
	payload: {
		startTimeInput,
	},
} );

export const setEndTimeInput = ( endTimeInput ) => ( {
	type: types.SET_END_TIME_INPUT,
	payload: {
		endTimeInput,
	},
} );

export const setSeparatorDate = ( separator ) => ( {
	type: types.SET_SEPARATOR_DATE,
	payload: {
		separator,
	},
} );

export const setSeparatorTime = ( separator ) => ( {
	type: types.SET_SEPARATOR_TIME,
	payload: {
		separator,
	},
} );

export const setAllDay = ( allDay ) => ( {
	type: types.SET_ALL_DAY,
	payload: {
		allDay,
	},
} );

export const setMultiDay = ( multiDay ) => ( {
	type: types.SET_MULTI_DAY,
	payload: {
		multiDay,
	},
} );

export const setTimeZone = ( timeZone ) => ( {
	type: types.SET_TIME_ZONE,
	payload: {
		timeZone,
	},
} );

export const setTimeZoneLabel = ( label ) => ( {
	type: types.SET_TIMEZONE_LABEL,
	payload: {
		label,
	},
} );

export const setTimeZoneVisibility = ( show ) => ( {
	type: types.SET_TIMEZONE_VISIBILITY,
	payload: {
		show,
	},
} );

export const setDateInputVisibility = ( show ) => ( {
	type: types.SET_DATE_INPUT_VISIBILITY,
	payload: {
		show,
	},
} );

export const allowEdits = () => ( {
	type: types.SET_DATETIME_BLOCK_EDITABLE_STATE,
	payload: {
		isEditable: true,
	},
} );

export const disableEdits = () => ( {
	type: types.SET_DATETIME_BLOCK_EDITABLE_STATE,
	payload: {
		isEditable: false,
	},
} );
