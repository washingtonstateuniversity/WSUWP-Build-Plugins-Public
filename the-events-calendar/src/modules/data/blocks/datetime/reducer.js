/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

export const DEFAULT_STATE = {
	start: momentUtil.toDateTime( defaultStartMoment ),
	end: momentUtil.toDateTime( defaultEndMoment ),
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguage: '',
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: date.FORMATS.TIMEZONE.string,
	timeZoneLabel: date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	showDateInput: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguage: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_LABEL:
			return {
				...state,
				timeZoneLabel: action.payload.label,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		case types.SET_DATE_INPUT_VISIBILITY:
			return {
				...state,
				showDateInput: action.payload.show,
			};
		default:
			return state;
	}
};
