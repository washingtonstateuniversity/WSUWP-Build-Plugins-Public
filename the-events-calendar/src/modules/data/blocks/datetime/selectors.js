/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const datetimeSelector = ( state ) => state.events.blocks.datetime;

export const getStart = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.start,
);

export const getEnd = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.end,
);

export const getStartTimeInput = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.startTimeInput,
);

export const getEndTimeInput = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.endTimeInput,
);

export const getAllDay = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.allDay,
);

export const getMultiDay = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.multiDay,
);

export const getDateSeparator = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.dateTimeSeparator,
);

export const getTimeSeparator = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.timeRangeSeparator,
);

export const getTimeZone = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.timeZone,
);

export const getTimeZoneVisibility = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.showTimeZone,
);

export const getTimeZoneLabel = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.timeZoneLabel,
);

export const getNaturalLanguageLabel = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.naturalLanguage,
);

export const getDateInputVisibility = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.showDateInput,
);

export const isEditable = createSelector(
	[ datetimeSelector ],
	( datetime ) => datetime.isEditable,
);
