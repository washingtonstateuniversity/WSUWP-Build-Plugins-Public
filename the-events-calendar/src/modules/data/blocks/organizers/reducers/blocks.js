/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
