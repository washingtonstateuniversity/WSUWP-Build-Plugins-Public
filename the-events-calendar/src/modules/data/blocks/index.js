/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
