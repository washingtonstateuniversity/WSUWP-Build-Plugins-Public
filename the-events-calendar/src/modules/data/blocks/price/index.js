/**
 * Internal dependencies
 */
import reducer from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import sagas from './sagas';

export default reducer;
export { selectors, actions, types, sagas };
