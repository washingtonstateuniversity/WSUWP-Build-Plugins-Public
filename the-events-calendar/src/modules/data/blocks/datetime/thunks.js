/**
 * Internal dependencies
 */
import {
	setStartDateTime,
	setEndDateTime,
	setStartTimeInput,
	setEndTimeInput,
	setAllDay as setAllDayAction,
	setMultiDay as setMultiDayAction,
	setSeparatorDate,
	setSeparatorTime,
	setTimeZone,
	setTimeZoneLabel,
	setTimeZoneVisibility,
	setNaturalLanguageLabel,
} from './actions';
import { DEFAULT_STATE } from './reducer';
import { maybeBulkDispatch } from '@moderntribe/events/data/utils';
import { date, moment } from '@moderntribe/common/utils';

const {
	isSameDay,
	parseFormats,
	toDateTime,
	toTime,
} = moment;

export const setInitialState = ( { get, attributes } ) => ( dispatch ) => {
	const timeZone = get( 'timeZone', DEFAULT_STATE.timeZone );
	const timeZoneLabel = get( 'timeZoneLabel', timeZone );

	/**
	 * @todo: remove maybeBuildDispatch, dispatch declaratively instead
	 */
	maybeBulkDispatch( attributes, dispatch )( [
		[ setStartDateTime, 'start', DEFAULT_STATE.start ],
		[ setEndDateTime, 'end', DEFAULT_STATE.end ],
		[ setAllDayAction, 'allDay', DEFAULT_STATE.allDay ],
		[ setSeparatorDate, 'separatorDate', DEFAULT_STATE.dateTimeSeparator ],
		[ setSeparatorTime, 'separatorTime', DEFAULT_STATE.timeRangeSeparator ],
		[ setTimeZone, 'timeZone', timeZoneLabel ],
		[ setTimeZoneLabel, 'timeZoneLabel', timeZoneLabel ],
		[ setTimeZoneVisibility, 'showTimeZone', DEFAULT_STATE.showTimeZone ],
	] );

	const values = {
		start: DEFAULT_STATE.start,
		end: DEFAULT_STATE.end,
	};

	if ( attributes.start ) {
		const startMoment = parseFormats( attributes.start );
		values.start = toDateTime( startMoment );
		const startTimeInput = toTime( startMoment );
		dispatch( setStartDateTime( values.start ) );
		dispatch( setStartTimeInput( startTimeInput ) );
	}
	if ( attributes.end ) {
		const endMoment = parseFormats( attributes.end );
		values.end = toDateTime( endMoment );
		const endTimeInput = toTime( endMoment );
		dispatch( setEndDateTime( values.end ) );
		dispatch( setEndTimeInput( endTimeInput ) );
	}

	dispatch( setNaturalLanguageLabel( date.rangeToNaturalLanguage( values.start, values.end ) ) );
	dispatch( setMultiDayAction( ! isSameDay( values.start, values.end ) ) );
};
