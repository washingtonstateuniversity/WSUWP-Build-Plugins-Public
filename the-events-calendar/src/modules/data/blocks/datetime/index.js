/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as thunks from './thunks';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, thunks, selectors, sagas };
