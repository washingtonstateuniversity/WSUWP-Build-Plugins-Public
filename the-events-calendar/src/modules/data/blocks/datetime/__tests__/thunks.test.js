/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';


/**
 * Internal dependencies
 */
import { thunks } from '@moderntribe/events/data/blocks/datetime';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

let initialState = {};
let store;

jest.mock( '@wordpress/data', () => ( {
	select: () => ( {
		isEditedPostNew: () => {},
	} ),
} ) );

describe( '[STORE] - Datetime thunks', () => {
	beforeEach( () => {
		initialState = {
			blocks: {
				datetime: {
					start: '2018-06-05 17:00:00',
					end: '2018-06-05 17:30:00',
					startTimeInput: '17:00',
					endTimeInput: '17:30',
					dateTimeSeparator: ' @ ',
					timeRangeSeparator: ' - ',
					allDay: false,
					multiDay: false,
					timezone: 'UTC',
				},
			},
		};

		store = mockStore( initialState );
	} );

	test( 'Initial set of state', () => {
		store = mockStore( {} );

		const attributes = {
			start: 'June 5, 2018 5:00 pm',
			end: 'June 5, 2018 5:30 pm',
			dateTimeSeparator: ' @ ',
			timeRangeSeparator: ' - ',
			allDay: false,
			multiDay: false,
			timeZone: 'UTC',
		};

		const get = jest.fn( ( name ) => attributes[ name ] );
		store.dispatch( thunks.setInitialState( { get, attributes } ) );

		expect( get ).toHaveBeenCalled();
		expect( get ).toHaveBeenCalledTimes( 2 );
		expect( store.getActions() ).toMatchSnapshot();
	} );
} );
