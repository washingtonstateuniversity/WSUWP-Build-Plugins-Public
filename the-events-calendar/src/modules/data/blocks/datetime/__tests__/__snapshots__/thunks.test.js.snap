// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Datetime thunks Initial set of state 1`] = `
Array [
  Object {
    "payload": Object {
      "start": "June 5, 2018 5:00 pm",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "end": "June 5, 2018 5:30 pm",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "allDay": false,
    },
    "type": "@@MT/EVENTS/SET_ALL_DAY",
  },
  Object {
    "payload": Object {
      "timeZone": "UTC",
    },
    "type": "@@MT/EVENTS/SET_TIME_ZONE",
  },
  Object {
    "payload": Object {
      "start": "2018-06-05 17:00:00",
    },
    "type": "@@MT/EVENTS/SET_START_DATE_TIME",
  },
  Object {
    "payload": Object {
      "startTimeInput": "5:00 pm",
    },
    "type": "@@MT/EVENTS/SET_START_TIME_INPUT",
  },
  Object {
    "payload": Object {
      "end": "2018-06-05 17:30:00",
    },
    "type": "@@MT/EVENTS/SET_END_DATE_TIME",
  },
  Object {
    "payload": Object {
      "endTimeInput": "5:30 pm",
    },
    "type": "@@MT/EVENTS/SET_END_TIME_INPUT",
  },
  Object {
    "payload": Object {
      "label": "June 5 2018 at 5:00 pm - 5:30 pm",
    },
    "type": "@@MT/EVENTS/SET_NATURAL_LANGUAGE_LABEL",
  },
  Object {
    "payload": Object {
      "multiDay": false,
    },
    "type": "@@MT/EVENTS/SET_MULTI_DAY",
  },
]
`;
